src/type_icons.c

```diff
@@ -473,13 +473,22 @@ static void FreeAllTypeIconResources(void)
 
 static bool32 ShouldHideTypeIcon(u32 battlerId)
 {
-    return gBattlerControllerFuncs[battlerId] != PlayerHandleChooseMove
-        && gBattlerControllerFuncs[battlerId] != HandleInputChooseMove
-        && gBattlerControllerFuncs[battlerId] != HandleChooseMoveAfterDma3
-        && gBattlerControllerFuncs[battlerId] != HandleInputChooseMove
-        && gBattlerControllerFuncs[battlerId] != HandleInputChooseTarget
-        && gBattlerControllerFuncs[battlerId] != HandleMoveSwitching
-        && gBattlerControllerFuncs[battlerId] != HandleInputChooseMove;
+	if (gBattlerControllerFuncs[battlerId] == PlayerHandleChooseMove)
+		return FALSE;
+	if (gBattlerControllerFuncs[battlerId] == HandleChooseMoveAfterDma3)
+		return FALSE;
+	if (gBattlerControllerFuncs[battlerId] == HandleInputChooseTarget)
+		return FALSE;
+	if (gBattlerControllerFuncs[battlerId] == HandleInputShowTargets)
+		return FALSE;
+	if (gBattlerControllerFuncs[battlerId] == HandleInputShowEntireFieldTargets)
+		return FALSE;
+	if (gBattlerControllerFuncs[battlerId] == HandleMoveSwitching)
+		return FALSE;
+	if (gBattlerControllerFuncs[battlerId] == HandleInputChooseMove)
+		return FALSE;
```

Comment:
would this work?
```
switch (gBattlerControllerFuncs[battlerId])
{
case PlayerHandleChooseMove:            return FALSE;
case HandleChooseMoveAfterDma3:         return FALSE;
case HandleInputChooseTarget:           return FALSE;
case HandleInputShowTargets:            return FALSE;
case HandleInputShowEntireFieldTargets: return FALSE;
case HandleMoveSwitching:               return FALSE;
case HandleInputChooseMove:             return FALSE;
}
```

Reply:
![image](https://github.com/user-attachments/assets/5feacba6-5936-499b-83cb-f9dbd6e15c29)
Compiler doesnt like that switch statement, @pkmnsnfrn your code does fix though

Reply:
a shame yours didn't work! 

https://github.com/rh-hideout/pokeemerald-expansion/pull/5201/commits/49089366b13144c851ce9ce173e12673bed28eae now just requires people add a single func to `showTypesControllerFuncs` and quits early if it finds a match

Reply:
please fix alignment

Reply:
https://github.com/rh-hideout/pokeemerald-expansion/pull/5201/commits/f94a7f3f80648db7f152ee5dde9419d2743923a8